ch_tools/common/commands/object_storage.py

```diff
@@ -74,30 +95,72 @@ def clean(
     Clean orphaned S3 objects.
     """
     disk_conf: S3DiskConfiguration = ctx.obj["disk_configuration"]
-    config = ctx.obj["config"]["object_storage"]["clean"]
+    config = ctx.obj["config"]["object_storage"]["space_usage"]
+    ch_client = clickhouse_client(ctx)
+
+    _cleanup_old_service_tables(ctx)
+
+    # Download cloud storage backups to include their blobs when collecting local blobs
+    downloaded_backups = _download_missing_cloud_storage_backups(
+        ctx, disk_conf.name, ignore_missing_cloud_storage_backups
+    )
```

Comment:
Backups can be wrapped in context manager as well, this will eliminate try-finally block completely